src/components/Badge/Badge.jsx

```diff
@@ -17,46 +17,18 @@ import OldBadges from './OldBadges';
 import BadgeReport from './BadgeReport';
 import { getUserProfile } from '../../actions/userProfile';
 
-const Badge = (props) => {
+const Badge = props => {
   const [isOpen, setOpen] = useState(false);
   const [isOpenTypes, setOpenTypes] = useState(false);
   const [totalBadge, setTotalBadge] = useState(0);
 
   const toggle = () => {
     if (isOpen) {
       const userId = props.userId;
-      props.getUserProfile(userId).then(() => {
-        let count = 0;
-        if (props.userProfile.badgeCollection) {
-          props.userProfile.badgeCollection.forEach((badge) => {
-            console.log('badge1', badge);
-            if (
-              badge?.badge?.badgeName === 'Personal Max' ||
-              badge?.badge?.type === 'Personal Max'
-            ) {
-              count += 1;
-            } else {
-              count += badge.count;
-            }
-          });
-          setTotalBadge(Math.round(count));
-        }
-      });
-    }
-    setOpen((isOpen) => !isOpen);
-  };
-
-  const toggleTypes = () => {
-    setOpenTypes((isOpenTypes) => !isOpenTypes);
-  };
-
-  useEffect(() => {
-    const userId = props.userId;
-    props.getUserProfile(userId).then(() => {
```

Comment:
I removed this because I don't see any association from this code. Also, it's the main reason behind accurate data being one refresh behind after login